@tgriesser A type error in the implementation is showing up on check-ts:

> src/cy/commands/actions/press.ts(76,32): error TS2345: Argument of type '(key: "Tab", userOptions?: (Partial<Loggable> & Partial<Timeoutable>) | undefined) => Promise<void>' is not assignable to parameter of type 'CommandFn<"press">'.
  Type 'Promise<void>' is not assignable to type 'void | Chainable<null>'.
    Type 'Promise<void>' is missing the following properties from type 'Chainable<null>': and, as, blur, check, and 94 more.

--- a/cli/CHANGELOG.md
+++ b/cli/CHANGELOG.md
@@ -10,6 +10,7 @@ _Released 5/20/2025 (PENDING)_
 **Misc:**
 
 - Cursor is now available as an IDE option for opening files in Cypress, if it is installed on your system. Addressed in [#31691](https://github.com/cypress-io/cypress/pull/31691).
+- `cy.press()` types now have a return type of `Chainable<null>` instead of `void`. Addressed in [#31698](https://github.com/cypress-io/cypress/pull/31698).
 
 **Dependency Updates:**
 

--- a/packages/driver/src/cy/commands/actions/press.ts
+++ b/packages/driver/src/cy/commands/actions/press.ts
@@ -36,7 +36,7 @@ export default function (Commands: Cypress.Commands, Cypress: Cypress.Cypress, c
 
       // throwErrByPath always throws, but there's no way to indicate that
       // code beyond this point is unreachable to typescript / linters
-      return
+      return null
     }
 
     if (Cypress.browser.family === 'webkit') {
@@ -47,7 +47,7 @@ export default function (Commands: Cypress.Commands, Cypress: Cypress.Cypress, c
         },
       })
 
-      return
+      return null
     }
 
     if (Cypress.browser.name === 'firefox' && Number(Cypress.browser.majorVersion) < 135) {
@@ -71,7 +71,11 @@ export default function (Commands: Cypress.Commands, Cypress: Cypress.Cypress, c
     } catch (err) {
       $errUtils.throwErr(err, { onFail: log })
     }
+
+    return null
   }
 
-  return Commands.add('press', pressCommand)
+  return Commands.addAll({
+    press: pressCommand,
+  })
 }

--- a/packages/driver/test/unit/cy/commands/actions/press.spec.ts
+++ b/packages/driver/test/unit/cy/commands/actions/press.spec.ts
@@ -51,9 +51,9 @@ describe('cy/commands/actions/press', () => {
       },
     }
 
-    // @ts-expect-error - this is a generic mock impl
     Commands = {
-      add: vi.fn(),
+      // @ts-expect-error - this is a generic mock impl
+      addAll: vi.fn(),
     }
 
     // @ts-expect-error
@@ -84,14 +84,14 @@ describe('cy/commands/actions/press', () => {
 
     addCommand(Commands, Cypress, cy, state, config)
 
-    expect(Commands.add).toHaveBeenCalledOnce()
+    expect(Commands.addAll).toHaveBeenCalledOnce()
 
     // @ts-expect-error
-    const [[cmdName, cmd]]: [[string, PressCommand]] = Commands.add.mock.calls
+    const [[obj]]: [[{press: PressCommand}]] = Commands.addAll.mock.calls
 
-    expect(cmdName).toEqual('press')
+    expect(typeof obj.press).toBe('function')
 
-    press = cmd as PressCommand
+    press = obj.press as PressCommand
   })
 
   describe('with a valid key', () => {

